src/lightning_app/cli/cmd_clusters.py

```diff
@@ -0,0 +1,238 @@
+import json
+import re
+import time
+from datetime import datetime
+
+import click
+from lightning_cloud.openapi import (
+    V1AWSClusterDriverSpec,
+    V1ClusterDriver,
+    V1ClusterPerformanceProfile,
+    V1ClusterSpec,
+    V1CreateClusterRequest,
+    V1InstanceSpec,
+    V1KubernetesClusterDriver,
+)
+from lightning_cloud.openapi.models import Externalv1Cluster, V1ClusterState, V1ClusterType
+from rich.console import Console
+from rich.table import Table
+from rich.text import Text
+
+from lightning_app.cli.core import Formatable
+from lightning_app.utilities.network import LightningClient
+from lightning_app.utilities.openapi import create_openapi_object, string2dict
+
+CLUSTER_STATE_CHECKING_TIMEOUT = 60
+MAX_CLUSTER_WAIT_TIME = 5400
+
+
+class AWSClusterManager:
+    """AWSClusterManager implements API calls specific to Lightning AI BYOC compute clusters when the AWS provider
+    is selected as compute backend."""
+
+    def __init__(self):
+        self.api_client = LightningClient()
+
+    def create(
+        self,
+        cost_savings: bool = False,
+        cluster_name: str = None,
+        role_arn: str = None,
+        region: str = "us-east-1",
+        external_id: str = None,
+        instance_types: [str] = [],
+        edit_before_creation: bool = False,
+        wait: bool = False,
+    ):
+        """request Lightning AI BYOC compute cluster creation.
+
+        :param cost_savings: flag indicating if the cluster should utilize cost savings mode
```

Comment:
Can we use Napoleon/google style?

Reply:
Yes, let's be consistent.

Reply:
Sure, I've changed it to Args: ... style